### PR TITLE
Add support for lines that end without a newline

### PR DESCRIPTION
**Pull request by Crestwave**
I also refactored the `head` function so it won't try to read another line after the limit is reached.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -393,10 +393,11 @@ Alternative to the `head` command.
 head() {
     # Usage: head "n" "file"
     while read -r line; do
-        [ "$i" = "$1" ] && break
         printf '%s\n' "$line"
         i=$((i+1))
+        [ "$i" = "$1" ] && return
     done < "$2"
```
Comment (dylanaraps) [inline README.md]: A a newline here.

File: README.md [hunk 2]
```diff
@@ -393,10 +393,11 @@ Alternative to the `head` command.
 head() {
     # Usage: head "n" "file"
     while read -r line; do
-        [ "$i" = "$1" ] && break
         printf '%s\n' "$line"
         i=$((i+1))
+        [ "$i" = "$1" ] && return
```
Comment (dylanaraps) [inline README.md]: I assume this works identically to before?

File: README.md (hunk 1 above)
Comment (Crestwave) [reply README.md]: Did you mean to select the `done` line?

File: README.md (hunk 2 above)
Comment (Crestwave) [reply README.md]: Other than the fact that it doesn't read an extra line, yes, as far as I can tell.

File: README.md (hunk 1 above)
Comment (dylanaraps) [reply README.md]: Yeah, my bad.

**Comment by dylanaraps** [comment]
LGTM :+1:

**Pull request merged**